### PR TITLE
Feature/combineEmail

### PR DESCRIPTION
**Pull request by sHuber125**
I would like to add these changes.  Discussed with Mike earlier today.  This combines all the email getting/login functions into a single class MailUtility.  I didn't yet, but Class/File Client.dart can be deleted, and class/file mailPiece_email_fetcher can be deleted.  I moved Client.dart feature into MailUtility (it just checks imapclient works with email/password for sign-in verification) and moved what I had in mailpiece_email_fetcher to mail_fetcher.

I also deleted any passing of email/username and password around app for the login function.  The username / password can be obtained from Keychain in the MailUtility function prior to logging in, so no need to pass between functions.

I also renamed a couple classes that had "mail" in the name but were really only working with mailPieces and storage.  I made it so only items with mail in the name are things that ask the MailUtility to get some emails and work on those emails.

### CODE REVIEW

File: lib/utility/Client.dart [hunk 1]
```diff
@@ -3,6 +3,7 @@ import 'package:enough_mail/enough_mail.dart';
 class Client {
   Client();
 
+  /* ScottH removed any use for this class...should be deleted.  Place all email login functions within MailUtility service
```
Comment (mdconatser) [inline lib/utility/Client.dart]: If this is actually not used at all anymore, can you just delete this?

File: lib/services/mailPiece_email_fetcher.dart [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
 class MailPieceEmailFetcher{
 
-  DateTime ? _timeStamp;
+  DateTime _timeStamp;
 
   /// Fetch the email using the mailPiece provided date
   /// from `uspsinformeddelivery@email.informeddelivery.usps.com`
   /// with the subject `Your Daily Digest`.
 
+  //Scott H is trying to add this to mail_fetcher.  No need for separate file.
```
Comment (mdconatser) [inline lib/services/mailPiece_email_fetcher.dart]: remove comment?

Comment (mdconatser) [inline lib/services/mailPiece_email_fetcher.dart]: I feel like the class structure here is still a little more complicated than it needs to be -- seems like mail_fetcher, mailPiece_email_fetcher, and mail_utility should really just be a single class to me. One class that represents retrieving data from the email server. Feel free to disregard this comment though if you disagree.

**Comment by shaneknows** [comment]
Please fix merge conflicts

**Comment by sHuber125** [comment]
> Please fix merge conflicts

I updated.  It could be because I had two or three PR at the same time there was a minor difference.

**Review by mdconatser — DISMISSED** [review]

File: lib/utility/Client.dart (hunk 1 above)
Comment (sHuber125) [reply lib/utility/Client.dart]: Yes, can delete.  I will delete and repush

File: lib/services/mailPiece_email_fetcher.dart (hunk 2 above)
Comment (sHuber125) [reply lib/services/mailPiece_email_fetcher.dart]: Yes, will delete this file and repush

Comment (sHuber125) [reply lib/services/mailPiece_email_fetcher.dart]: they could be combined.  I think let's leave Mail Utility as a separate class though

**Review by mdconatser — APPROVED** [review]

**Review by shaneknows — APPROVED** [review]

**Pull request merged**